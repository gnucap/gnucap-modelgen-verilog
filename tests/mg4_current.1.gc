
attach ./modelgen_0.so


verilog

`modelgen
module test_current1(p, n, c);
	electrical p, n, c;
	inout p, n, c;
	(* desc="" *) real i;
	analog begin
		I(p, n) <+ 1.;
		i = I(p, n);
		V(c) <+ i;
	end
endmodule

!make test_current1.so > /dev/null
attach ./test_current1.so

parameter v=0

test_current1 dut(1,0,2);
resistor #(1) v1(1, 0);

list

print dc i(dut) v(nodes)

dc v 0 1 .5 trace=i
end
