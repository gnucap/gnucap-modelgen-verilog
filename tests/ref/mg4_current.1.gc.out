parameter v=0 
test_current1 #() dut (.p(1),.n(0),.c(2));
resistor #(.r(1)) v1 (.p(1),.n(0));
#           i(dut)     v(1)       v(2)      
 0.         0.         0.         0.        
-1.         1.        -1.         1.        
-2.         1.        -1.         1.        
 0.         1.        -1.         1.        
 0.         1.        -1.         1.        
-1.         1.        -1.         1.        
 0.5        1.        -1.         1.        
 0.         1.        -1.         1.        
-1.         1.        -1.         1.        
 1.         1.        -1.         1.        
